nullaway/src/test/resources/com/uber/nullaway/testdata/CheckFieldInitPositiveCases.java

```diff
@@ -156,7 +156,7 @@ static void init() {}
 
   static class T11 {
 
-    // BUG: Diagnostic contains: @NonNull static field f not initialized
+    // BUG: Diagnostic contains: @NonNull static field T11.f not initialized
     static Object f;
 
     static {
```

Comment:
Can we get an Anonymous class with uninitialized fields test case for this? (see comment above)

Reply:
@lazaroclapp @msridhar 
`getQualifiedName` outputs fully qualified class name by connecting `.` (not `com.example.Foo$Bar$Baz` but `com.example.Foo.Bar.Baz`). So I think it is difficult to outputs  `Foo.Bar.Baz` by finding  last `.` . 

I can get `Foo.Bar.Baz` with following steps as an alternative idea.
1. get outermost fully qualified class name by using `outermostClass` method.(e.g. `com.example.Foo`)
2. get outermost class name by finding last `.`. (e.g. `Foo`)
3. get fully qualified class name by using `getQualifiedName` method.(e.g. `com.example.Foo.Bar.Baz`)
4. get all of the enclosing classes name by using `indexOf("Foo")` and `substring` method 

Do you have any problems with this?

I will add anonymous class with uninitialized fields test case.

Reply:
I recently had a similar requirement. I recall that `ClassSymbol#flatName()` does return the name-with-dollar-signs. (If you don't already have a reference to the relevant `ClassSymbol`, try to retrieve it using `ASTHelpers.getSymbol(ClassTree)`.)

Reply:
That seems to be the case from the [documentation](https://javadoc.io/doc/org.kohsuke.sorcerer/sorcerer-javac/latest/com/sun/tools/javac/code/Symbol.ClassSymbol.html#flatname). @shtke can you try `flatName()` first?

Reply:
@Stephan202 @lazaroclapp 
I tried `flatName()` and retrieved the name-with-dollar-signs !  Thank you for the advice.